unit-tests/py/rspy/devices.py

```diff
@@ -114,9 +164,9 @@ def by_name( name ):
     """
     sns = set()
     global _device_by_sn
-    for sn, dev in _device_by_sn.items():
-        if dev.supports( rs.camera_info.name ) and dev.get_info( rs.camera_info.name ).find( name ) >= 0:
-            sns.add( sn )
+    for device in _device_by_sn.values():
+        if device.name  and  device.name.find( name ) >= 0:
```

Comment:
This will not work if the name is the product line name

Reply:
If name == device.name, find() will return 0 and we're good, no?

Reply:
Yes.